docs/500_ENCODER.md

```diff
@@ -0,0 +1,50 @@
+# What is an Encoder?
+The responsibility of an encoder is to interpret raw `PointState` values and
+provide specific operations that cannot be done purely on the raw values. One
+example would be validation of `PointState` raw values.
+
+An `Encoder` should ideally not contain any state and should simply be an
+interpreter. All functions should be pure.
+
+An `Encoder` can be used by an `Observer`, but in that case the `Observer`
+and the `Plugin` should share the same `Encoder`. An `Encoder` typically maps to
+a single contract instance or registry.
+
+Ideally an encoder should not maintain any state and should be able to be 
+represented with static functions instead of reference functions.
```

Comment:
can we enforce this? I don't know about go programming paradigms, but in other languages encode would be an abstract class instead of an object

Reply:
maybe this makes more sense as a util library?

Reply:
Yes, but it wouldn't be very goish. In most cases a function would just accept an interface and the implementation would decide if it was a static or pointer reference. A check **could** be done to ensure the struct passed in matching the interface was not a pointer. I would not advise this as it really seems like a strong anti-pattern.